assembly/pallets/balances/balances.ts

```diff
@@ -38,40 +41,62 @@ export class Balances {
         if(BalancesConfig.existentialDeposit().unwrap() > freeBalance.unwrap()) {
             return ResponseCodes.VALIDITY_ERROR;
         }
-        BalancesStorageEntries.Account().set(freeBalance, accountId);
+        const accData = BalancesStorageEntries.Account().get(accountId);
+        accData.setFree(freeBalance);
+        accData.setReserved(reservedBalance);
+        BalancesStorageEntries.Account().set(accData, accountId);
         Balances._depositEvent(EventTypes.BalanceSet, accountId.toU8a().concat(freeBalance.toU8a()));
         return ResponseCodes.SUCCESS;
     }
 
+    /**
+     * @description Dummy transfer call
+     * @param dest 
+     * @param value 
+     * @returns 
+     */
+    static transfer(dest: AccountId, value: Balance): u8[] {
+        return [];
+    }
+
     /**
      * @description Transfer the given value from source to destination
      * @param source source account
      * @param dest dest account
      * @param value value of the transfer
      */
-    static transfer(source: AccountIdType, dest: AccountIdType, value: Balance): u8[] {
+    static _applyTransfer(source: AccountIdType, dest: AccountIdType, value: Balance): u8[] {
         const senderBalance = BalancesStorageEntries.Account().get(source);
         const receiverBalance = BalancesStorageEntries.Account().get(dest);
 
-        if(senderBalance.unwrap() < value.unwrap()) {
+        if(senderBalance.getFree().unwrap() < value.unwrap()) {
             return ResponseCodes.INSUFFICIENT_BALANCE;
         }
 
-        const senderNewBalance = senderBalance.unwrap() - value.unwrap();
-        const receiverNewBalance = receiverBalance.unwrap() + value.unwrap();
-
         // Make sure new balances is higher than ExistentialDeposit
-        if(senderNewBalance < BalancesConfig.existentialDeposit().unwrap()) {
+        //@ts-ignore
+        if(senderBalance.getFree().unwrap() - value.unwrap() < BalancesConfig.existentialDeposit().unwrap()) {

```

Comment:
Lets extract this into a variable:
`senderBalance.getFree().unwrap() - value.unwrap()` 
since we are using it 2 times